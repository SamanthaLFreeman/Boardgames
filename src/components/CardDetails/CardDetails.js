import React from 'react';
import { connect } from 'react-redux';

export const CardDetails = ({ name, year_published, min_players, max_players, max_playtime, description_preview, image_url, primary_publisher, categoriesIds, rules_url, categoriesKey}) => {
  const displayCategories = categoriesIds.map(category => category.id).reduce((acc, id) => {
    categoriesKey.forEach(key => {
      if(id === key.id) {
        acc.push(key.name)
      }
    })
    return acc;
  }, []).map(categoryName => <li key={categoryName}>{categoryName}</li>)
  return (
    <section>
      <img src={image_url} alt={name} />
      <h3>{name}</h3>
      <p>Year Published: {year_published}</p>
      <p>Min Players: {min_players}</p>
      <p>Max Players: {max_players}</p>
      <p>Max Playtime: {max_playtime}</p>
      <p>{description_preview}</p>
      <p>Primary Publisher: {primary_publisher}</p>
      <p>Categories:</p>
      <ul>
        {displayCategories}
      </ul>
      <p>Rules: {rules_url}</p>
    </section>
  )
}

export const mapStateToProps = state => ({
  categoriesKey: state.categories
})

export default connect(mapStateToProps, null)(CardDetails);