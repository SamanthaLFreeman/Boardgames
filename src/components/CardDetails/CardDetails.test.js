import React from 'react';
import { shallow } from 'enzyme';
import { CardDetails } from './CardDetails';

describe('CardDetails', () => {
  let wrapper;
  const mockCard = {}
  const categoriesIdsMock = [
    {
      id: 1234
    }
  ]
  const categoriesKeyMock = [
    {
      id: 1234,
      name: "Fantasy"
    }
  ]

  beforeEach(() => {
    wrapper = shallow(<CardDetails 
    card={mockCard}
    categoriesIds={categoriesIdsMock}
    categoriesKey={categoriesKeyMock} />)
  });

  it('should match the snapshot with the data passed through', () => {
    expect(wrapper).toMatchSnapshot();
  });
})