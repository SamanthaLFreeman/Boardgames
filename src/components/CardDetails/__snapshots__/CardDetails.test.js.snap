// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardDetails should match the snapshot with the data passed through 1`] = `
<section>
  <img />
  <h3 />
  <p>
    Year Published: 
  </p>
  <p>
    Min Players: 
  </p>
  <p>
    Max Players: 
  </p>
  <p>
    Min Playtime: 
  </p>
  <p>
    Max Playtime: 
  </p>
  <p />
  <p>
    Primary Publisher: 
  </p>
  <p>
    Categories:
  </p>
  <ul>
    <li
      key="Fantasy"
    >
      Fantasy
    </li>
  </ul>
  <p>
    Rules: 
  </p>
</section>
`;
